core/src/operations/decorators/mod.rs

```diff
@@ -59,3 +72,16 @@ impl<'a> DecoratorIterator<'a> {
         }
     }
 }
+
+impl<'a> Iterator for DecoratorIterator<'a> {
```

Comment:
Is this used anywhere?

Reply:
Yes, we use this [here](https://github.com/0xPolygonMiden/miden-vm/blob/fc7b247e28b640ea6206c86efc9044e2041f4f25/processor/src/lib.rs#L386).